pkg/pool-linear/contracts/LinearPool.sol

```diff
@@ -605,19 +601,34 @@ contract LinearPool is BasePool, IGeneralPool, LinearMath, IRateProvider {
         // but being stricter makes analysis easier at little expense.
 
         (uint256 currentLowerTarget, uint256 currentUpperTarget) = getTargets();
-        _require(
-            balances[_mainIndex] >= currentLowerTarget && balances[_mainIndex] <= currentUpperTarget,
-            Errors.OUT_OF_TARGET_RANGE
-        );
-
-        _require(
-            balances[_mainIndex] >= newLowerTarget && balances[_mainIndex] <= newUpperTarget,
-            Errors.OUT_OF_NEW_TARGET_RANGE
-        );
+        _require(_isMainBalanceWithinTargets(currentLowerTarget, currentUpperTarget), Errors.OUT_OF_TARGET_RANGE);
+        _require(_isMainBalanceWithinTargets(newLowerTarget, newUpperTarget), Errors.OUT_OF_NEW_TARGET_RANGE);
 
         _setTargets(_mainToken, newLowerTarget, newUpperTarget);
     }
 
+    function setSwapFeePercentage(uint256 swapFeePercentage) public override {
+        // For the swap fee percentage to be changeable:
+        //  - the pool must currently be between the current targets (meaning no fees are currently pending)
+        //
+        // As the amount of accrued fees is not explicitly stored but rather derived from the main token balance and the
+        // current swap fee percentage, requiring for no fees to be pending prevents the fee setter from changing the
+        // amount of pending fees, which they could use to e.g. drain Pool funds in the form of inflated fees.
+
+        (uint256 lowerTarget, uint256 upperTarget) = getTargets();
+        _require(_isMainBalanceWithinTargets(lowerTarget, upperTarget), Errors.OUT_OF_TARGET_RANGE);
+
+        super.setSwapFeePercentage(swapFeePercentage);
+    }
+
+    function _isMainBalanceWithinTargets(uint256 lowerTarget, uint256 upperTarget) private view returns (bool) {
+        bytes32 poolId = getPoolId();
+        (, uint256[] memory balances, ) = getVault().getPoolTokens(poolId);
+        uint256 mainTokenBalance = _upscale(balances[_mainIndex], _scalingFactor(_mainToken));
```

Comment:
Also could test the UNHANDLED_BY_LINEAR_POOL exit (e.g., try to exit when unpaused with a non-proportional exit kind).

Reply:
Sorry, I don't follow. What OUT_OF_BOUNDS?

re. the gas savings, we're not currently struggling for bytecode, and your change might be misleading. That property only holds for main token: for the wrapped token `_scalingFactor()` does not equal the state variable - I don't want to make it seems like it might.

Reply:
For the OUT_OF_BOUNDS, there's an edge case code path that's not triggered by explicit tests. Like the test itself, not really necessary, but easy to add.

```
        // Sanity check: this is not entirely necessary as the Vault's interface enforces the indices to be valid, but
        // the check is cheap to perform.
        _require(indexIn < _TOTAL_TOKENS && indexOut < _TOTAL_TOKENS, Errors.OUT_OF_BOUNDS);
```